packages/next/server/body-streams.ts

```diff
@@ -69,7 +73,11 @@ export function getClonableBody<T extends IncomingMessage>(
      */
     async finalize(): Promise<void> {
       if (buffered) {
-        await endPromise
+        const res = await endPromise
+
+        if (res && typeof res === 'object' && res.error) {
```

Comment:
Would this be possible as well?

```suggestion
        if (res?.error) {
```

Reply:
The more verbose checks is meant for type assertions